de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/configuration/RepositoryConfiguration.java

```diff
@@ -139,4 +140,34 @@ public String getLocalPath() {
 	public void setLocalPath(String localPath) {
 		properties.setProperty(LOCAL_PATH_KEY, localPath);
 	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((properties == null) ? 0 : properties.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+		RepositoryConfiguration other = (RepositoryConfiguration) obj;
+		if (properties == null) {
+			if (other.properties != null) {
+				return false;
+			}
+		} else if (!properties.equals(other.properties)) {
+			return false;
+		}
+		return true;

```

Comment:
This segment looks weird. What if properties is null but other.properties is not null?
I think we can fix and simplify this via

```suggestion
		return Objects.equals(properties, other.properties)

```